targets/liteos-stm32f4/Makefile

```diff
@@ -0,0 +1,38 @@
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+JSLIB ?= targets/liteos-stm32f4/bin
+
+# path to the LiteOS base directory
+LITEOSBASE ?= $(CURDIR)/../LiteOS_Kernel
+# path to the JERRYSCRIPT directory
+JERRYDIR ?= $(CURDIR)
+
+# path to the application directory
+APPDIR ?= $(JERRYDIR)/targets/liteos-stm32f4/source
+
+# path to the binary directory
+BINDIR ?= $(JERRYDIR)/targets/liteos-stm32f4/bin/
+
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+INCLUDES += -I$(JERRYDIR)/jerry-core/include
+
+
+# Add the jerry lib
+JS_CFLAGS ?= -L $(JSLIB) -ljerrycore
```

Comment:
I've been trying to get my head around this makefile, and I'm very confused now. APPDIR, BINDIR, and INCLUDES are variables that are not used anywhere. (Really. Not in this makefile and not even in the (transitively) included files.) However, the included makefile, the one that seems to be a support file to build LiteOS + user apps for a Discovery board (I mean, seemingly a general purpose makefile for all user apps out there) contains JerryScript-specific parts(!!!). E.g., all the source directories of jerry-core are hard-coded to be include paths, and the sources of this target (which is not even in master but in this PR only) are hard-coded to be added to the list of sources -- in a makefile in the LiteOS repository. Moreover, this JS_CFLAGS is passed to the compiler in that makefile. See https://github.com/LITEOS/LiteOS_Kernel/blob/b29d4036a353ce8f05fd45ae4284ece44d5a310e/projects/STM32F429I_DISCO_GCC/Makefile .

To me, this is unacceptable as it is unmaintainable. There is no way that anyone will update the directory structure of jerry-core or targets in LiteOS, should they change anytime. The kernel/OS project should not contain app-specific parts but should contain parts that enable the extension/override of the defaults by the user app or parts that complement the user app. App-side makefile should be able to set LDFLAGS, CFLAGS, ASFLAGS, SRC (to name a few), that declare what the app needs and the OS-side complement should extend those settings.

Notes:
- the `:=` initialization of SRC and ASM_SRC prevent user apps to extend the list of compiled/assembled files determined in the OS makefile.
- the name JS_CFLAGS is a misnomer as it is used at linking (anyway, it should go away).
- a JerryScript application should not try to access headers deep within jerry-core, neither from the OS-side makefile nor from the app-side makefile -- it should only access the public headers in jerry-core/include.